File: src/main/java/com/summer/lijiahao/abs/AbstractItemListener.java
package com.summer.lijiahao.abs;

import java.awt.event.ItemEvent;
import java.awt.event.ItemListener;

/**
 * δΈζηε¬
 */
public abstract class AbstractItemListener implements ItemListener {

    private final AbstractDialog dialog;

    public AbstractItemListener(AbstractDialog dialog) {
        this.dialog = dialog;
    }

    @Override
    public void itemStateChanged(ItemEvent e) {
        afterSelect(e);
    }

    public abstract void afterSelect(ItemEvent e);

    public AbstractDialog getDialog() {
        return dialog;
    }
}

File: src/main/kotlin/com/summer/lijiahao/nccdevtool/devconfig/listener/base/AbstractItemListener.kt
package com.summer.lijiahao.nccdevtool.devconfig.listener.base

import com.summer.lijiahao.nccdevtool.devconfig.ui.NCCloudDevConfigDialog
import java.awt.event.ItemListener

abstract class AbstractItemListener(dialog: NCCloudDevConfigDialog) : ItemListener {
    var dialog: NCCloudDevConfigDialog

    init {
        this.dialog = dialog
    }
}